src/app/modules/payments/stripe.controller.ts

```diff
@@ -295,7 +296,107 @@ export const checkPaymentReceiptStatus: ControllerHandler<{
     })
 }
 
-// TODO: Refactor for use in static payment url implementation
+/**
+ * Handler for GET /api/v3/payments/:formId/:paymentId/invoice/download
+ * Receives Stripe webhooks and updates the database with transaction details.
+ *
+ * @returns 200 if webhook is successfully processed
+ * @returns 404 if the PaymentId is not found
+ * @returns 404 if the FormId is not found
+ * @returns 404 if payment.completedPayment?.receiptUrl is not found
+ */
+export const downloadPaymentInvoice: ControllerHandler<{
+  formId: string
+  paymentId: string
+}> = (req, res) => {
+  const { formId, paymentId } = req.params
+  logger.info({
+    message: 'downloadPaymentInvoice endpoint called',
+    meta: {
+      action: 'downloadPaymentInvoice',
+      formId,
+      paymentId,
+    },
+  })
+
+  return ResultAsync.combine([
+    PaymentService.findPaymentById(paymentId),
+    FormService.retrieveFullFormById(formId).andThen(checkFormIsEncryptMode),
+  ])
+    .map(([payment, populatedForm]) => {
+      logger.info({
+        message: 'Found paymentId in payment document',
+        meta: {
+          action: 'downloadPaymentInvoice',
+          payment,
+        },
+      })
+      if (!payment.completedPayment?.receiptUrl) {
+        return res
+          .status(StatusCodes.NOT_FOUND)
+          .send({ message: 'Receipt url not ready' })
+      }
+      // retrieve receiptURL as html
+      return (
+        axios
+          .get<string>(payment.completedPayment.receiptUrl)
+          // convert to pdf and return
+          .then((receiptUrlResponse) => {
+            const html = receiptUrlResponse.data
+            const businessInfo = populatedForm.admin.agency.business
+
+            // we will still continute the invoice generation even if there's no address/gstregno
+            if (!businessInfo.address || !businessInfo.gstRegNo)
+              logger.warn({
+                message:
+                  'Some business info not available during invoice generation',
+                meta: {
+                  action: 'downloadPaymentInvoice',
+                  payment,
+                  agencyName: populatedForm.admin.agency.fullName,
+                  businessInfo: businessInfo,
+                },
+              })
+            const invoiceHtml = convertToInvoiceForrmat(html, {
+              address: businessInfo.address || '',
```

Comment:
Empty string default will make the invoice look weird, but this is an error to correct so I think that's fine.